FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -28,26 +27,37 @@ class Flagsmith constructor(
     private val baseUrl: String = "https://edge.api.flagsmith.com/api/v1",
     private val context: Context? = null,
     private val enableAnalytics: Boolean = DEFAULT_ENABLE_ANALYTICS,
-    private val analyticsFlushPeriod: Int = DEFAULT_ANALYTICS_FLUSH_PERIOD_SECONDS
+    private val enableCache: Boolean = DEFAULT_ENABLE_CACHE,
+    private val cacheTTLSeconds: Long = DEFAULT_CACHE_TTL_SECONDS,
+    private val analyticsFlushPeriod: Int = DEFAULT_ANALYTICS_FLUSH_PERIOD_SECONDS,
+    private val defaultFlags: List<Flag> = emptyList()
 ) {
-    private val client: FlagsmithClient = FlagsmithClient(baseUrl, environmentKey)
+    private val retrofit: FlagsmithRetrofitService = FlagsmithRetrofitService.create(baseUrl, environmentKey, cacheTTLSeconds, context, enableCache)
     private val analytics: FlagsmithAnalytics? =
         if (!enableAnalytics) null
-        else if (context != null) FlagsmithAnalytics(context, client, analyticsFlushPeriod)
+        else if (context != null) FlagsmithAnalytics(context, retrofit, analyticsFlushPeriod)
         else throw IllegalArgumentException("Flagsmith requires a context to use the analytics feature")
 
+    init {
+        if (enableCache && context == null) {
+            throw IllegalArgumentException("Flagsmith requires a context to use the cache feature")
+        }
+    }
+
     companion object {
         const val DEFAULT_ENABLE_ANALYTICS = true
+        const val DEFAULT_ENABLE_CACHE = true
```

Comment:
I would recommend that we don't set the default to true here, otherwise we would need to create a major release version.

Reply:
Fair enough, now disabled by default